randovania/exporter/hints/credits_spoiler.py

```diff
@@ -67,40 +67,69 @@ def generic_credits(
     standard_pickup_configuration: StandardPickupConfiguration,
     all_patches: dict[int, GamePatches],
     players_config: PlayersConfiguration,
-    namer: HintNamer,
-    pickup_name_format: str = "{}",
-    use_player_color: bool = True,
-) -> dict[str, str]:
+) -> dict[PickupEntry, list[OwnedPickupLocation | str]]:
+    """
+    Returns the credits in the form of dictionary that can be used to create your own formatting.
+    The key will be a PickupEntry, and the value a list of location entries. An entry is either an OwnedPickupLocation
+    if the pickup is located somewhere, or a string which is used for special cases (such as starting with the pickup).
+    """
     major_pickup_name_order = {
         pickup.name: index for index, pickup in enumerate(standard_pickup_configuration.pickups_state.keys())
     }
 
-    def sort_pickup(p: PickupEntry) -> tuple[float, str]:
+    def sort_pickup(tup: tuple[PickupEntry, list[OwnedPickupLocation | str]]) -> tuple[float, str]:
+        p = tup[0]
         return major_pickup_name_order.get(p.name, math.inf), p.name
 
-    details = get_locations_for_major_pickups_and_keys(all_patches, players_config)
-    major_pickups_spoiler = {
-        pickup: [entry.export(namer, use_player_color) for entry in entries] for pickup, entries in details.items()
-    }
+    details: dict[PickupEntry, list[OwnedPickupLocation | str]] = get_locations_for_major_pickups_and_keys(
+        all_patches, players_config
+    )
 
     starting_pickups = starting_pickups_with_count(all_patches[players_config.player_index])
     for pickup, count in starting_pickups.items():
         if count < 1:
             continue
 
-        if pickup not in major_pickups_spoiler:
-            major_pickups_spoiler[pickup] = []
```

Comment:
@henriquegemignani These lines cause it to add minor pickups to the spoiler. Was that intended? I currently left the functionality but got surprised seeing missile and pb tanks showing up in the credits

Reply:
These lines in particular? So the missiles don't exist in `major_pickups_spoiler`, but exists in `starting_pickups`?

Reply:
That is the case yes. I'm asking if  it was intentional whether minor pickups should be listed in the credits if they're random starting items. It feels weird since the credits usually only ever has majors.

Reply:
Cool, then i'll make a seperate PR for that.